ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/SqlSequence.scala

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.dao.events
+
+import com.daml.platform.store.SimpleSqlAsVectorOf._
+import anorm.{ResultSetParser, Row, RowParser, SimpleSql}
+import scalaz.{-\/, Free, Functor, \/-}
+
+import java.sql.Connection
+
+object SqlSequence {
+
+  /** A sequence of `SimpleSql`s, terminating in A. */
+  type T[A] = Free[Elt, A]
+
+  // this representation is just trampolined Reader, but exposing that
+  // would be unsound because it is _not_ distributive, and anyway
+  // we may want to make the representation more explicit for more complex
+  // analysis (e.g. applying polymorphic transforms to all contained SimpleSqls...)
+  final class Elt[+A] private[SqlSequence] (private[SqlSequence] val run: Connection => A)
```

Comment:
What does `Elt` stand for? Can you spell it out, please? I believe it would be more readable.

Reply:
It's "element"; sure.

Reply:
Ah, thanks!

Reply:
a8c62b02698866cb8b3914822a9cee8b519a3b31